aws/smithy-aws-apigateway-openapi/src/main/java/software/amazon/smithy/aws/apigateway/openapi/AddAuthorizers.java

```diff
@@ -15,75 +15,120 @@
+                Authorizer authorizer = entry.getValue();
+                if (converter.getAuthSchemeName().equals(scheme)) {
+                    SecurityScheme createdScheme = converter.createSecurityScheme(context);
+                    SecurityScheme.Builder schemeBuilder = createdScheme.toBuilder();
+                    schemeBuilder.putExtension(
+                            CLIENT_EXTENSION_NAME, determineApiGatewayClientName(authorizer.getScheme()));
 
-        SecurityScheme.Builder builder = securityScheme.toBuilder();
+                    ObjectNode authorizerNode = Node.objectNodeBuilder()
+                            .withMember("type", authorizer.getType())
+                            .withMember("authorizerUri", authorizer.getUri())
+                            .withOptionalMember("authorizerCredentials", authorizer.getCredentials().map(Node::from))
+                            .withOptionalMember("identityValidationExpression",
+                                                authorizer.getIdentityValidationExpression().map(Node::from))
+                            .withOptionalMember("identitySource", authorizer.getIdentitySource().map(Node::from))
+                            .withOptionalMember("authorizerResultTtlInSeconds",
+                                                authorizer.getResultTtlInSeconds().map(Node::from))
+                            .build();
+                    schemeBuilder.putExtension(EXTENSION_NAME, authorizerNode);
 
-        // The client authorization method is not contained within the authorizer in
-        // the OpenAPI extension.
-        authorizer.getClientType().ifPresent(type -> builder.putExtension(CLIENT_EXTENSION_NAME, type));
+                    LOGGER.fine(() -> String.format("Adding the `%s` OpenAPI security scheme", entry.getKey()));
+                    components.putSecurityScheme(entry.getKey(), schemeBuilder.build());
+                    break;
+                }
+            }
+        }
 
-        ObjectNode authorizerNode = Node.objectNodeBuilder()
-                .withMember("type", authorizer.getType())
-                .withMember("authorizerUri", authorizer.getUri())
-                .withOptionalMember("authorizerCredentials", authorizer.getCredentials().map(Node::from))
-                .withOptionalMember("identityValidationExpression",
-                                    authorizer.getIdentityValidationExpression().map(Node::from))
-                .withOptionalMember("identitySource", authorizer.getIdentitySource().map(Node::from))
-                .withOptionalMember("authorizerResultTtlInSeconds", authorizer.getResultTtlInSeconds().map(Node::from))
-                .build();
+        builder.components(components.build());
+        return builder.build();
+    }
 
-        return builder.putExtension(EXTENSION_NAME, authorizerNode).build();
+    // TODO: should this also allow overrides via configuration properties?
```

Comment:
Is this configuration override necessary for having the [`x-amazon-apigateway-authtype` end up as `oauth2`](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-authtype.html)?

Reply:
Yes, and any other potential values that we might need to support